### DISCUSSION TITLE
Limited compatibility with Unprompted extension.

### DISCUSSION THREAD

**Discussion by BlastedRemnants** [adieyal/sd-dynamic-prompts#249] [General]
I wasn't sure how to describe this, or even if I should mention it here or on the Unprompted page instead, so I'll probably post over there as well. In any case I've had occasional issues with the Combinatorial generation feature in Dynamic Prompts, and after much frustration and experimenting I've found that it will very rarely work properly when Unprompted is enabled, but with Unprompted disabled it works flawlessly every time. I'm using up to date versions of Auto and both extensions, with manual reinstalls of both Unprompted and Dynamic Prompts yesterday. Figured I'd mention it because it took me a few hours to sort out for myself, and none of the solutions offered here helped me, so if anyone else has the same problem they might find a fix.
Edit: forgot to mention that by enable/disable Unprompted I mean with the checknox in the UI, shown in the screenshot below.
![Screenshot (22)](https://user-images.githubusercontent.com/72319405/217701448-de9781ce-9e64-4b61-b10f-7ae0e98d477b.png)

**Comment by adieyal** [comment]
I haven't used Unprompted but from what I understand, there is overlapping functionality, perhaps even overlapping syntax. What are you getting from running them both at the same time? I can understand that both extensions have functionality that the other doesn't, but is there a situation where you need to use both? Perhaps either DP or Unprompted should include that functionality so that you don't need to run both simultaneously.

**Reply by BlastedRemnants** [reply]
The two main functions from Unprompted that I use are the txt2mask feature and the ability to load a random checkpoint at generation time. The checkpoint loading is done with what the Unprompted dev calls shortcodes, and it let me use a wildcard as input so I was happy to finally have found a way to do random checkpoints. For the txt2mask part I could use the older stand-alone script but as far as I know there's no function in Dynamic Prompts that lets me load a random checkpoint. It's not a big deal either way, the only real problem I have is that I can't do random models and combinatorial batches at the same time, everything else seems to cooperate fine.

**Reply by adieyal** [reply]
There have been discussions about letting dynamic prompts change settings dynamically including loading checkpoints - I previously felt it was out of scope but I've been convinced that it makes sense. Re txt2mask, it's good that you found a solution. In general, I'm hesitant to implement features from other extensions.

It might be worth a discussion with the ThereforeGames crew to see if there is a way for the two extensions to co-exist.

**Reply by BlastedRemnants** [reply]
Thanks for taking the time to respond to this at all, I wasn't sure if I should even mention it since it isn't really an issue or problem, and even if it was then it would be on Unprompted's end. Speaking for myself I'd love to see the ability to randomize checkpoints and perhaps other settings with Dynamic Prompts, the wildcard lists give us great control and variety at the same time and that's fantastic I'm sure but you don't want to get too cluttered with features either. In any case this is probably my most used extension, keep up the great work!! I'll probably start a discussion over on Unprompted, should probably at least let them know that there is some conflict between your extension and there's but in the meantime it's fairly rare that I want to do combinatorial generations and load random checkpoints at the same time so it's not a big deal. For you tho I guess it's just another thing to keep in mind if you have a user that mentions their combinatorial batches aren't working, could be something to remember if you're helping someone troubleshoot. Anyway tho thanks again, cheers!
